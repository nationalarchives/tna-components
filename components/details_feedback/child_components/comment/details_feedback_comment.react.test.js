import React from 'react';
import TestRenderer from 'react-test-renderer';
import Comment from './details_feedback_comment.react';

let testRenderer, testRendererToJson;

// Snapshot test 1
it('renders the checkbox component correctly', () => {
  const items = ['Comment Label'];
  const tree = TestRenderer.create(
    <Comment commentLabel={items[0]} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

// Snapshot test 2
it('renders the checkbox component with optional props', () => {
  const items = [
    'Hello world',
    'this-is-the-id',
    () => console.log('Function test'),
    'Comment warning'
  ];
  const tree = TestRenderer.create(
    <Comment
      id={items[1]}
      commentLabel={items[0]}
      onChange={items[2]}
      commentWarning={items[3]}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

// Unit test
describe("Check DOM element and it's attributes", () => {
  const items = ['Comment Label'];
  testRenderer = TestRenderer.create(<Comment commentLabel={items[0]} />);

  testRendererToJson = testRenderer.toJSON();
  it('Should have an <div> tag', () => {
    expect(testRendererToJson.type).toBe('div');
  });
  it('Should have a class', () => {
    expect(testRendererToJson.props.className).toBe('comment');
  });
  it('Should have children', () => {
    expect(testRendererToJson.children.length).toBe(2);
    expect(testRendererToJson.children[0].type).toBe('label');
    expect(testRendererToJson.children[1].type).toBe('textarea');
  });
  it('Should have a label tag with props', () => {
    expect(
      testRendererToJson.children[0].props.hasOwnProperty('for')
    ).toBeTruthy();
  });
  it('Should have a input tag with props', () => {
    expect(
      testRendererToJson.children[1].props.hasOwnProperty('onChange')
    ).toBeTruthy();
    expect(
      testRendererToJson.children[1].props.hasOwnProperty('name')
    ).toBeTruthy();
    expect(
      testRendererToJson.children[1].props.hasOwnProperty('rows')
    ).toBeTruthy();
    expect(
      testRendererToJson.children[1].props.hasOwnProperty('cols')
    ).toBeTruthy();
    expect(
      testRendererToJson.children[1].props.hasOwnProperty('id')
    ).toBeTruthy();
    expect(
      testRendererToJson.children[1].props.hasOwnProperty('autoFocus')
    ).toBeTruthy();
  });
});
