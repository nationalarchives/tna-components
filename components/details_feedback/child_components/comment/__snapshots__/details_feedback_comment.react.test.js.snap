// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the checkbox component correctly 1`] = `
<div
  className="comment"
>
  <label
    for={null}
  >
    Comment Label
  </label>
  <input
    autoFocus={null}
    id={null}
    onChange={null}
    type="text"
  />
</div>
`;

exports[`renders the checkbox component with optional props 1`] = `
<div
  className="comment"
>
  <label
    for="this-is-the-id"
  >
    Hello world
  </label>
  <input
    autoFocus={null}
    id="this-is-the-id"
    onChange={[Function]}
    type="text"
  />
  <p
    className={null}
  >
    Comment warning
  </p>
</div>
`;
